### PR TITLE
Fix const node non-deterministic names (part 1)

### PR DESCRIPTION
**Pull request by achetver**
Fix const node non-deterministic names in `model-optimizer\extensions\front` and `model-optimizer\mo`

Code:

* [x] Comments
* [x] Code style (PEP8)
* [x] Transformation generates reshape-able IR
* [x] Transformation preserves node names

Validation:

* [x] Unit tests: Not needed - no new functions implemented
* [x] Framework layer tests: Not needed - no new layers were enabled in the MO
* [x] Transformation tests: Not needed - no new transformations implemented
* [x] e2e model test with an update: Not needed
* [x] MO IR Reader check: Not needed

Documentation:

* [x] Supported frameworks operations list: Not needed - no new supported operations
* [x] Supported public models list: Not needed - no new supported models
* [x] New operations specification: Not needed - no new operations implemented
* [x] Guide on how to convert the public model: Not needed - no new supported models
* [x] User guide update: Not needed

Other:

* [x] Sample/Demo application to infer the model: Not needed

### CODE REVIEW

File: model-optimizer/extensions/front/kaldi/add_reshape_around_pooling.py [hunk 1]
```diff
@@ -49,38 +49,41 @@ def pattern(self):
...
+                                             {'name': node_name + '/concat_all_dims', 'in_ports_count': 4, 'axis': 0})
 
-        concat = Concat(graph, {'in_ports_count': 4, 'axis': 0}).create_node()
         concat.in_port(0).connect(split.out_port(0))
         concat.in_port(3).connect(mul.out_port(0))
-        concat.in_port(2).connect(const_1.out_port(0))
         concat.in_port(1).connect(node_pool_stride.out_port(0))
 
-        reshape_in = Reshape(graph, {'name': '/Reshape/' + node.name}).create_node()
+        reshape_in = Reshape(graph, {'name': '/Reshape/' + node_name}).create_node()
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/kaldi/add_reshape_around_pooling.py:81]: Please, change name to `node_name + '/reshape_in'`

File: model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py [hunk 2]
```diff
@@ -40,11 +40,9 @@ def run_before(self):
 
     def replace_op(self, graph: Graph, node: Node):
         # split input to (i_part, f_part, c_part, o_part, ct_1)
-        split_node_axis = Const(graph, {'value': np.int64(1)}).create_node()
-        split_node = Split(graph, {'name': 'Split_lstm_input_',
-                                   'num_splits': 5}).create_node()
+        split_node = create_op_with_const_inputs(graph, Split, {1: np.int64(1)}, {'name': 'Split_lstm_input_',
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py:43]: Change node name to `node.soft_get('name', node.id) + '/split_lstm_input'`

File: model-optimizer/extensions/front/mxnet/gather.py [hunk 3]
```diff
@@ -26,10 +26,10 @@ class GatherFrontReplacer(FrontReplacementOp):
 
     def replace_sub_graph(self, graph: Graph, match: dict):
         node = match['op']
-        gather_node = Gather(graph, dict(name=node.id + '/embedding_',
-                                         symbol_dict={'name': node.id + '/embedding_'})).create_node()
-        axis_const = Const(graph, {'value': int64_array(0)}).create_node()
+
+        gather_node = create_op_with_const_inputs(graph, Gather, {2: int64_array(0)}, {'name': node.id + '/embedding_',
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/mxnet/gather.py:30]: Please, do not use `node.id`. Use regular `node.soft_get('name', node.id)` instead. 

Also, get rid of `symbol_dict` from the code. It should not be used at all.

File: model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py [hunk 4]
```diff
@@ -51,28 +51,30 @@ def replace_sub_graph(self, graph: Graph, match: dict):
...
         else:
             shape = Shape(graph, {'name': name + '/input_shape'}).create_node()
 
             idxs = list(range(axis)) if axis > 0 else list(range(axis, 0))
 
             axis_shape_portion = node_to_get_shape_value_of_indices(shape, idxs)
             first_dims = create_op_node_with_second_input(graph, ReduceProd, int64_array([0]),
-                                                          {'keep_dims': True})
-            second_dims = Const(graph, {'value': int64_array([-1])}).create_node()
+                                                          {'name': name + '/first_dims', 'keep_dims': True})
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py:67]: there is no variable called `name` in this scope. Looks like this code is not covered in the pre-commit.

File: model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py [hunk 5]
```diff
@@ -51,28 +51,30 @@ def replace_sub_graph(self, graph: Graph, match: dict):
...
+            second_dims = Const(graph, {'value': int64_array([-1]), 'name': name + '/second_dims'}).create_node()
 
             node.in_port(0).get_source().connect(shape.in_port(0))
             axis_shape_portion.out_port(0).connect(first_dims.in_port(0))
 
             order_of_dims = [first_dims, second_dims] if axis > 0 else [second_dims, first_dims]
 
             dim = new_shape_node_from_shape_nodes(order_of_dims)
+            dim.name = reshape_node.name + '/shape'
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py:76]: This is not a right solution. Could the function `new_shape_node_from_shape_nodes` set appropriate name?

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py [hunk 6]
```diff
@@ -41,8 +41,9 @@ def replace_op(self, graph: Graph, node: Node):
                       }
         indexes_node = Const(graph).create_node(attrs=const_attrs)
 
-        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64}).create_node()
-        axis_const = Const(graph, {'value': int64_array(0)}).create_node()
+        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64,
+                                  'name': 'input_order/{}'.format(node.id)}).create_node()
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/kaldi/extractors/copy_ext.py:45]: Change name to `node.soft_get('name', node.id) + '/order'`

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py [hunk 7]
```diff
@@ -41,8 +41,9 @@ def replace_op(self, graph: Graph, node: Node):
...
+        axis_const = Const(graph, {'value': int64_array(0), 'name': 'axis/{}'.format(node.id)}).create_node()
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/kaldi/extractors/copy_ext.py:46]: Use `create_op_node_with_second_input` to create `gather_node` node. The `axis_const` is used in the Gather op only.

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py [hunk 8]
```diff
@@ -41,8 +41,9 @@ def replace_op(self, graph: Graph, node: Node):
                       }
         indexes_node = Const(graph).create_node(attrs=const_attrs)
 
-        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64}).create_node()
-        axis_const = Const(graph, {'value': int64_array(0)}).create_node()
+        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64,
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/kaldi/extractors/copy_ext.py:44]: Please, update names to `gather_node`, `perm1_node ` and `perm2_node` nodes.

File: model-optimizer/mo/graph/perm_inputs.py [hunk 9]
```diff
@@ -56,9 +56,11 @@ def axis(op_node: Node, port_info: str, input_port: int):
 
     data_node = op_node.in_node(input_port)
 
-    const = Const(graph, {'value': permutation.inv, 'need_shape_inference': True}).create_node_with_data()
-    axis_const = Const(graph, {'value': int64_array(0)}).create_node_with_data()
-    gather = Gather(graph, {'name': op_node.name + '/AxisGather', 'need_shape_inference': True}).create_node_with_data(
+    gather_name = op_node.name + '/AxisGather'
```
Comment (lazarevevgeny) [inline model-optimizer/mo/graph/perm_inputs.py:59]: `op_node.soft_get('name', op_node.id) + '/AxisGather'`

File: model-optimizer/mo/graph/perm_inputs.py [hunk 10]
```diff
@@ -103,14 +105,18 @@ def order(op_node: Node, port_info: str, input_port: int):
 
     data_node = op_node.in_node(input_port)
 
-    const = Const(graph, {'value': permutation.perm, 'need_shape_inference': True}).create_node_with_data()
-    axis_const = Const(graph, {'value': int64_array(0)}).create_node_with_data()
-    gather = Gather(graph, {'name': op_node.name + '/OrderGather_1',
+    gather_name = op_node.name + '/OrderGather_1'
```
Comment (lazarevevgeny) [inline model-optimizer/mo/graph/perm_inputs.py:108]: The same comment as above

File: model-optimizer/extensions/front/kaldi/add_reshape_around_pooling.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/extensions/front/kaldi/add_reshape_around_pooling.py]: Done.

File: model-optimizer/extensions/front/mxnet/gather.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/extensions/front/mxnet/gather.py]: Done.

File: model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py (hunk 4 above)
Comment (achetver) [reply model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py:67]: line 49: `name = node.soft_get('name', node.id)`

File: model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py (hunk 5 above)
Comment (achetver) [reply model-optimizer/extensions/front/onnx/flattenONNX_to_reshape.py]: Function new_shape_node_from_shape_nodes updated to set name for created node.

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py (hunk 7 above)
Comment (achetver) [reply model-optimizer/mo/front/kaldi/extractors/copy_ext.py]: Done.

File: model-optimizer/mo/graph/perm_inputs.py (hunk 9 above)
Comment (achetver) [reply model-optimizer/mo/graph/perm_inputs.py]: Done.

File: model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py (hunk 2 above)
Comment (achetver) [reply model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py]: Done.

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py (hunk 6 above)
Comment (achetver) [reply model-optimizer/mo/front/kaldi/extractors/copy_ext.py]: Done.

File: model-optimizer/mo/graph/perm_inputs.py (hunk 10 above)
Comment (achetver) [reply model-optimizer/mo/graph/perm_inputs.py]: Done.

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py (hunk 8 above)
Comment (achetver) [reply model-optimizer/mo/front/kaldi/extractors/copy_ext.py:47]: Done.

File: model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py [hunk 11]
```diff
@@ -40,11 +40,10 @@ def run_before(self):
...
-        split_node = Split(graph, {'name': 'Split_lstm_input_',
-                                   'num_splits': 5}).create_node()
+        split_node = create_op_with_const_inputs(graph, Split, {1: np.int64(1)},
+                                                 {'name': node.soft_get('name', node.id) + '/split_lstm_input',
+                                                  'num_splits': 5})
         node.in_port(0).get_connection().set_destination(split_node.in_port(0))
-        split_node.in_port(1).connect(split_node_axis.out_port(0))
 
         # i_t = Sigmoid(i_part + w_ic*ct_1)
         i_scale_attrs = {'name': 'i_scaleshift',
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py:49]: Update name for `i_scale`, `sum_i_c`, `i_sigmoid`, `f_scale_attrs` and others below

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py [hunk 12]
```diff
@@ -33,26 +34,27 @@ def replace_op(self, graph: Graph, node: Node):
...
                        'value': np.array(weights),
                        'shape': [weights_size],
                        'data_type': np.int32
                       }
         indexes_node = Const(graph).create_node(attrs=const_attrs)
 
-        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64}).create_node()
-        axis_const = Const(graph, {'value': int64_array(0)}).create_node()
-        perm1_node = Transpose(graph, {'name': 'input_permute'}).create_node([node.in_node(0)])
+        perm_in_1 = Const(graph, {'value': np.array([1, 0], dtype=np.int64), 'shape': [2], 'data_type': np.int64,
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/kaldi/extractors/copy_ext.py:47]:
```
        perm_in_1 = Const(graph, {'value': int64_array([1, 0]), 'name': node_name + '/order'}).create_node()
```

File: model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py (hunk 11 above)
Comment (achetver) [reply model-optimizer/extensions/front/kaldi/replace_lstm_nonlinearity.py]: Done.

File: model-optimizer/mo/front/kaldi/extractors/copy_ext.py (hunk 12 above)
Comment (achetver) [reply model-optimizer/mo/front/kaldi/extractors/copy_ext.py]: Fixed.

**Comment by achetver** [comment]
@jane-intel Can you, please, take a look at these changes?

**Pull request merged**